static/js/stream_list.js

```diff
@@ -402,12 +512,8 @@ exports.rename_stream = function (sub) {
 
 $(function () {
     $(document).on('narrow_activated.zulip', function (event) {
-        if (zoomed_to_topics && (active_stream_name() !== zoomed_stream)) {
-            zoom_out();
-        }
 
-        $("ul.filters li").removeClass('active-filter active-subject-filter');
-        remove_expanded_subjects();
+        reset_to_unnarrowed();
```

Comment:
This block seems problematic, because `reset_to_unnarrowed` doesn't contain the `active_stream_name` check.  I think you could fix this like this:
```
function reset_to_unnarrowed(narrowed_within_stream) {
    if (zoomed_to_topics && narrowed_within_stream !== true) {
        zoom_out();
    }
   ...
```